rfcs/067-prismic-api-ids/README.md

```diff
@@ -0,0 +1,46 @@
+# RFC 067: Prismic API ID casing
+
+## TL;DR
+- Use kebab-case for Custom type API IDs (plural/singular for reusable/single respectively)
+- Use camelCase for all (Custom type and Slice) field API IDs
+- Use snake_case for Slice API IDs
+
+## Background
+We’d like to make sure our Prismic API IDs are written with consistent casing and that we’re following defaults from Prismic so that we’re going with the grain as far as possible and not having to manually override auto-generated SliceMachine files.
+
+## Custom type API IDs
+When we started using Prismic we decided that these should be kebab-cased. Also, if the type was ‘reusable’ it should be plural, if it was 'single' it should be singular. For example, the reusable 'Exhibition highlight tour' type has an API ID of `exhibition-highlight-tours` whereas the single 'Global alert' type has an API ID of `global-alert`.
+
+SliceMachine makes these snake_case by default but allows this to be overridden easily in the Graphical User Interface (GUI) and to-date we have made *all* of them kebab-case, so there probably isn't a good case for changing these to snake_case now.
+
+A couple of reusable content types have been given singular API IDs – `card` and `collection-venue` – should we consider changing these two to `cards` and `collection-venues`?
+
+## Field API IDs
+Field API IDs are individual properties on a Custom type or Slice. More often than not these are individual words (e.g. `title`), which keeps things simple. When there's more than one word, we now have a mixture of snake_case and camelCase field API IDs.
+
+- [Example of a camelCase field](https://github.com/wellcomecollection/wellcomecollection.org/blob/main/common/views/slices/CollectionVenue/model.json#L37)
+- [Example of a snake_case field](https://github.com/wellcomecollection/wellcomecollection.org/blob/main/common/views/slices/GuideStop/model.json#L31)
+
+SliceMachine defaults to snake_case (based on the label), although this is readily overridable in the GUI at the point of adding the field.
+
+We exclusively work with camelCased variables in TypeScript and while snake_cased object properties aren’t a problem (e.g. `data.some_property`), we wouldn’t be able to destructure those properties off the `data` object without it being a linting error:
+
+```
+const { some_property } = data; // <-- linting error
+```
+
+Whereas this isn’t a problem:
+
+```
+const { someProperty } = data; // <-- ✨
+```
+
+So perhaps the preference should be for updating the field names at point of creation to be camelCased to remove the need for an extra renaming variable step in the TypeScript.
```

Comment:
I feel like since it's "readily overridable in the GUI at the point of adding the field.", we should enforce camelCase for sure, again, as a documented agreed upon decision.

Could we also consider aligning existing ones with the migration script?

Reply:
I think we should consider it – I added a new vs legacy para at the end to wave my hands in the air about whether or not we want to do it based on cost/benefit